ui/tests/unit/serializers/ldap/role-test.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupTest } from 'vault/tests/helpers';
+
+module('Unit | Serializer | ldap/role', function (hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function () {
+    const store = this.owner.lookup('service:store');
+    this.model = store.createRecord('ldap/role', {
+      backend: 'ldap',
+      name: 'test-role',
+      dn: 'cn=hashicorp,ou=Users,dc=hashicorp,dc=com',
+      rotation_period: '24h',
+      username: 'hashicorp',
+      creation_ldif: 'foo',
+      deletion_ldif: 'bar',
+      rollback_ldif: 'baz',
+      username_template: 'default',
+      default_ttl: '1h',
+      max_ttl: '24h',
+    });
+  });
+
+  test('it should serialize attributes based on type', async function (assert) {
+    assert.expect(11);
+
+    const serializeAndAssert = (type) => {
+      this.model.type = type;
+      const payload = this.model.serialize();
+
+      assert.strictEqual(
+        Object.keys(payload).length,
+        this.model.fieldsForType.length,
```

Comment:
Personally I prefer to be more explicit in tests, so that even if the calculated value changes we can see on the test output exactly what it expected before, vs what it has now. I don't think it's that much more work to update a test when the behavior changes, and not having to update a test when there is functionality added is a code smell

Reply:
Do you mean not using `fieldsForType` in this case and hard coding the same arrays in the test? I think that's a good idea in case `fieldsForType` changes unintentionally and little work to update if fields are added and the test needs to be updated.